File: frontend/lib/core/models/meal_skip.dart
class MealSkip {
  final String id;
  final String userId;
  final DateTime date;
  final String mealType;
  final DateTime createdAt;

  const MealSkip({
    required this.id,
    required this.userId,
    required this.date,
    required this.mealType,
    required this.createdAt,
  });

  factory MealSkip.fromJson(Map<String, dynamic> json) => MealSkip(
        id: json['id'] as String,
        userId: json['user_id'] as String,
        date: DateTime.parse(json['date'] as String),
        mealType: json['meal_type'] as String,
        createdAt: DateTime.parse(json['created_at'] as String),
      );
}

File: backend/app/schemas/meal_skip.py
import uuid
from datetime import date, datetime

from pydantic import BaseModel, ConfigDict

from app.models.meal_skip import MealType


class SkipCreate(BaseModel):
    date: date
    meal_type: MealType


class AdminSkipOverride(BaseModel):
    user_id: uuid.UUID
    date: date
    meal_type: MealType


class SkipResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    user_id: uuid.UUID
    date: date
    meal_type: MealType
    created_at: datetime
